app/models/status.rb

```diff
@@ -303,12 +303,26 @@ def favourites_count
     status_stat&.favourites_count || 0
   end
 
+  # Reblogs count received from an external instance
+  def untrusted_reblogs_count
+    status_stat&.untrusted_reblogs_count unless local?
+  end
+
+  # Favourites count received from an external instance
+  def untrusted_favourites_count
+    status_stat&.untrusted_favourites_count unless local?
+  end
+
   def increment_count!(key)
     update_status_stat!(key => public_send(key) + 1)
+    increment_count!(:untrusted_favourites_count) if key == :favourites_count && !untrusted_favourites_count.nil?
```

Comment:
This will issue up to 3 different `UPDATE`s from a single method call, I'm also uncomfortable with the recursivity here.

Reply:
gotcha. wrote it this way just to try and minimize changes to the method, but i can rewrite it to batch

Reply:
At the very least this could be:

```ruby
def increment_count!(key)
  if key == :favourites_count && !untrusted_favourites_count.nil?
    update_status_stat!(favourites_count: favourites_count + 1, untrusted_favourites_count: untrusted_favourites_count + 1)
  elsif key == :reblogs_count && !untrusted_reblogs_count.nil?
    # Equivalent...
  else
    update_status_stat!(key => public_send(key) + 1)
  end
end
```

Reply:
ah i just did this here https://github.com/mastodon/mastodon/pull/32620/commits/29729167b77d728682ada4c79e18a68ef59ec768

but i like that better, one sec

Reply:
that's here - https://github.com/mastodon/mastodon/pull/32620/commits/fcd970170e42cceccf5d9c69471af5cceb0a1d5c

doesn't feel very DRY but i think moving those methods to the counter model/making a counter concern would be the next step if we expected linked values like this to happen again